app/assets/javascripts/plugins_toggle.js

```diff
@@ -40,7 +40,7 @@ const setupOnce = require('setup_once');
     },
 
     toggleState: function(){
-      this.state = !this.state;
+      this.state = this.state === 'published' ? 'unpublished' : 'published';
```

Comment:
using 'published' and 'unpublished' instead of booleans seems like a good change